src/solver/optimisation/opt_gestion_des_bornes_cas_lineaire.cpp

```diff
@@ -394,21 +394,36 @@ void OPT_InitialiserLesBornesDesVariablesDuProblemeLineaire(PROBLEME_HEBDO* Prob
                 if (AllMustRunGeneration->AllMustRunGenerationOfArea[Pays] > 0.)
                     MaxAllMustRunGenerationOfArea
                       = AllMustRunGeneration->AllMustRunGenerationOfArea[Pays];
-
+                
                 C = C + MaxAllMustRunGenerationOfArea;
                 if (C >= 0.)
                     Xmax[Var] = C + 1e-5;
                 else
                     Xmax[Var] = 0.;
 
+                //adq patch: update ENS <= DENS
+                if(ProblemeHebdo->UseAdequacyPatch == true)
+                {
+                    if(ProblemeHebdo->AdequacyFirstStep == true)
+                    {
+                        ProblemeHebdo->ResultatsHoraires[Pays]->ValeursHorairesDENS[PdtHebdo] = 0.0; //adq patch init if 1st run
+                    }
+                    else if(ProblemeHebdo->AdequacyFirstStep == false) //2nd run update upper bound with DENS
+                    {
+                        if(ProblemeHebdo->ResultatsHoraires[Pays]->ValeursHorairesDENS[PdtHebdo] >= 0.)
+                        {
+                            Xmax[Var] = ProblemeHebdo->ResultatsHoraires[Pays]->ValeursHorairesDENS[PdtHebdo] + 1e-5;
+                            // logs.debug() << "opt_gestion_bornes:" << Pays << ":" << PdtHebdo << ":Xmax[Var] = " << Xmax[Var];
+                        }
+                        else
+                            Xmax[Var] = 0.;
+                    }
+                }
```

Comment:
Creating test cases when realized.
From specifications: "additional constraint is introduced in the optimization problem for physical nodes that are declared in the adequacy patch domain (nodes declared as “2” in “adequacy_patch.ini” file". 
Am I missing something ? or I do not see if loop where ENS<= DENS only if area type is 2?

Reply:
Nice. Fixed